go/vt/sqlparser/predicate_rewriting.go

```diff
@@ -16,35 +16,72 @@ limitations under the License.
 
 package sqlparser
 
+import (
+	"log"
+)
+
 const (
 	Changed  RewriteState = true
 	NoChange RewriteState = false
 )
 
 type RewriteState bool
 
+var logger *log.Logger
+
+// Set☝️to something not nil to get debug output
+// var logger *log.Logger = log.New(os.Stdout, "", 0)
+
+func f(e Expr) func() Expr {
+	return func() Expr {
+		return e
+	}
+}
```

Comment:
It may be useful to either change the name of the function or add a comment explaining why the function is doing this.